service/controller/terminate_unhealthy_node.go

```diff
@@ -0,0 +1,192 @@
+	"github.com/giantswarm/k8sclient/v5/pkg/k8sclient"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	"github.com/giantswarm/operatorkit/v4/pkg/controller"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/metricsresource"
+	"github.com/giantswarm/operatorkit/v4/pkg/resource/wrapper/retryresource"
+	"github.com/giantswarm/tenantcluster/v3/pkg/tenantcluster"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"sigs.k8s.io/cluster-api/api/v1alpha3"
+
+	"github.com/giantswarm/azure-operator/v5/client"
+	"github.com/giantswarm/azure-operator/v5/pkg/credential"
+	"github.com/giantswarm/azure-operator/v5/pkg/label"
+	"github.com/giantswarm/azure-operator/v5/pkg/project"
+	"github.com/giantswarm/azure-operator/v5/service/collector"
+	"github.com/giantswarm/azure-operator/v5/service/controller/resource/terminateunhealthynode"
+)
+
+type TerminateUnhealthyNodeConfig struct {
+	K8sClient k8sclient.Interface
+	Logger    micrologger.Logger
+
+	AzureMetricsCollector collector.AzureAPIMetrics
+	CredentialProvider    credential.Provider
+	SentryDSN             string
+}
+
+type TerminateUnhealthyNode struct {
+	*controller.Controller
+}
+
+func NewTerminateUnhealthyNode(config TerminateUnhealthyNodeConfig) (*controller.Controller, error) {
+	var err error
+
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+	if config.AzureMetricsCollector == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.AzureMetricsCollector must not be empty", config)
+	}
+	if config.CredentialProvider == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.CredentialProvider must not be empty", config)
+	}
+
+	var certsSearcher *certs.Searcher
+	{
+		c := certs.Config{
+			K8sClient: config.K8sClient.K8sClient(),
+			Logger:    config.Logger,
+
+			WatchTimeout: 30 * time.Second,
+		}
+
+		certsSearcher, err = certs.NewSearcher(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var resources []resource.Interface
+	{
+		resources, err = newTerminateUnhealthyNodeResources(config, certsSearcher)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var operatorkitController *controller.Controller
+	{
+		c := controller.Config{
+			InitCtx: func(ctx context.Context, obj interface{}) (context.Context, error) {
+				return ctx, nil
+			},
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+			// Name is used to compute finalizer names. This results in something
+			// like operatorkit.giantswarm.io/azure-operator-machine-pool-controller.
+			Name: project.Name() + "-terminate-unhealthy-node-controller",
+			NewRuntimeObjectFunc: func() runtime.Object {
+				return new(v1alpha3.Cluster)
+			},
```

Comment:
Does `operatorkit` allow creating controllers that don't watch any CRD? That way it will work effectively like a cronjob, and we would know that it would be executed every X minutes.

Reply:
but we need a crd in this case, beacuse the check has to be executed in the tenant cluster not the control plane